.github/workflows/github-cxx-qt-tests.yml

```diff
@@ -6,41 +6,219 @@
-      # Install clang-format, Qt, reuse, valgrind
-      - run: sudo apt-get update
-      - run: sudo apt-get install -y clang-format-12 libssl-dev pkg-config python3-pip qtbase5-dev qtdeclarative5-dev qtquickcontrols2-5-dev qml-module-qtquick2 qml-module-qtquick-controls2 qml-module-qttest valgrind
-      - run: sudo ln -s /usr/bin/clang-format-12 /usr/local/bin/clang-format
-      - run: pip3 install reuse
-      # Ensure we have a clean build folder
-      - run: rm -rf build/
-      - run: mkdir -p build/
-      # Configure and build CMake then run ctest
-      #
-      # Note that each github action run command is executed from the same directory even after
-      # performing cd so for the ctest line we need to cd within the same command
-      - run: cmake -S. -Bbuild/ -G "Unix Makefiles"
-      - run: cmake --build build/ --config Debug --target all -j$(nproc)
-      - run: (cd build/ && QT_QPA_PLATFORM=offscreen ctest -j$(nproc) -C Debug -T test --output-on-failure)
+    - name: "Clone Git repository"
+      uses: LebedevRI/checkout@issue197
+      with:
+        path: ${{ matrix.workspace }}
+    - name: "Install Rust toolchain"
+      uses: actions-rs/toolchain@v1
+      with:
+        components: clippy, rustfmt
+        toolchain: stable
+
+    - name: "Rust tools cache"
+      uses: actions/cache@v2
+      id: rust-tools-cache
+      with:
+        path: |
+          ${{ matrix.cargo_dir }}/bin/sccache${{ matrix.exe_suffix }}
+          ${{ matrix.cargo_dir }}/bin/mdbook${{ matrix.exe_suffix }}
+          ${{ matrix.cargo_dir }}/bin/mdbook-linkcheck${{ matrix.exe_suffix }}
+        key: ${{ matrix.os }}_sccache-0.2.15_mdbook-0.4.15_mdbook-linkcheck-0.2.8
+
+    - name: "Build Rust tools"
+      if: steps.rust-tools-cache.outputs.cache-hit != 'true'
+      run: cargo install sccache mdbook mdbook-linkcheck
+
+    - name: "Compiler cache"
+      uses: actions/cache@v2
+      with:
+        path: ${{ matrix.compiler_cache_path }}
+        key: ${{ matrix.name }}-${{ github.head_ref }}-${{ github.run_number }}
+        restore-keys: |
+          ${{ matrix.name }}-${{ github.head_ref }}
+          ${{ matrix.name }}
+
+    - name: "[Ubuntu] Install dependencies"
+      if: runner.os == 'Linux'
+      run: >-
+        sudo add-apt-repository ${{ matrix.ppa }} &&
+        sudo apt-get update &&
+        sudo apt-get install -y
+        ninja-build
+        clang-format-12
+        libssl-dev
+        pkg-config
+        python3-pip
+        valgrind
+        ${{ matrix.packages-extra }}
+    - name: "[macOS] Install dependencies"
+      if: runner.os == 'macOS'
+      # automake is needed for building libicu which is a dependency of Qt
+      run: |
+        brew install automake ninja clang-format
+    - name: "Install reuse"
+      if: runner.os == 'Linux'
+      run: pip3 install reuse
+
+    - name: "Authenticate to GitHub Packages"
+      # TODO: replace 'Be-ing' with 'KDAB'
```

Comment:
Note that this TODO and the TODO in CMakeLists.txt for generating a PAT will need to be taken care of by you after merging this. I can't push vcpkg packages to KDAB's GitHub Packages because this is a PR from a forked repository so GITHUB_TOKEN does not have write permission.

You could merge this PR as is then take care of this afterwards. Alternatively, you could merge/cherry pick the commits besides the one introducing vcpkg, then cherry pick the vcpkg commit and address the TODOs in one modified commit.

Reply:
Ah I've found where to create personal access tokens, but for my account. I'll need one for the organisation...

Reply:
> Generate a read-only token form somewhere (where do i do this? sweat_smile ) for the kdab packages?

Yes, do that here: https://github.com/settings/tokens
Be sure to not paste the entire string verbatim into CMakeLists.txt and read the comments there.

> The GITHUB_TOKEN in github-cxx-qt-tests I just leave and is it auto generated ? or do i need to add this as a "secret" into the settings?

Yes, GITHUB_TOKEN is generated automatically by GitHub Actions. You do not need to manually add a secret. https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

> Then once I have all of that, i commit it and push directly into the main branch, and watch CI run on the commit and build the packages (will take a while).

You don't need to do it on the main branch. You could push to a branch on the upstream (KDB/cxx-qt) repo rather than your user's fork. And yes it will take like 3-4 hours.

> Then future pull requests should pick up the read-only one from CMake ?

Pull requests from forks and well as local builds will have read-only access. When you want to update dependencies, simply merge the master branch of https://github.com/microsoft/vcpkg into the vcpkg submodule and push that either to the main branch or another branch on the upstream repository.

Reply:
Or it seems like you can't for orgs, but if your user account has permission for that org then you can use tokens you generate. So a token i create might be enough.

Reply:
> Ah I've found where to create personal access tokens, but for my account. I'll need one for the organisation...

I don't think they exist at the organization level. Generating one for your user is fine. Make sure to set it to not expire automatically after a time limit when you create it.

Reply:
Cool, I see thanks! Just double checking with the org owners that they are OK with this setup :-)

Reply:
It may help for them to know that GitHub Packages is free for public repositories and there are no documented limitations on storage space or bandwidth: https://github.com/features/packages#pricing (look at the left side, not the table)